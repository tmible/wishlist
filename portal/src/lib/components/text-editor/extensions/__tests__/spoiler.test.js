import { describe, expect, it, vi } from 'vitest';
import Spoiler from '../spoiler.js';

describe('text editor / extensions / spoiler', () => {
  it('should be defined', () => {
    expect(Spoiler).toMatchSnapshot();
  });

  it('should parse HTML', () => {
    expect(Spoiler.config.parseHTML()).toMatchSnapshot();
  });

  it('should render HTML', () => {
    expect(Spoiler.config.renderHTML()).toMatchSnapshot();
  });

  describe('add commands', () => {
    it('should add set command', () => {
      const setMark = vi.fn();
      Spoiler.config.addCommands().setSpoiler()({ commands: { setMark } });
      expect(setMark).toHaveBeenCalledWith(Spoiler.name);
    });

    it('should add toggle command', () => {
      const toggleMark = vi.fn();
      Spoiler.config.addCommands().toggleSpoiler()({ commands: { toggleMark } });
      expect(toggleMark).toHaveBeenCalledWith(Spoiler.name);
    });

    it('should add unset command', () => {
      const unsetMark = vi.fn();
      Spoiler.config.addCommands().unsetSpoiler()({ commands: { unsetMark } });
      expect(unsetMark).toHaveBeenCalledWith(Spoiler.name);
    });
  });

  describe('add keyboard shortcuts', () => {
    it('should add keyboard shortcuts', () => {
      expect(Spoiler.config.addKeyboardShortcuts()).toMatchSnapshot();
    });

    it('should implement keyboard shortcut', () => {
      const toggleSpoiler = vi.fn();
      const bindValue = { editor: { commands: { toggleSpoiler } } };
      Spoiler.config.addKeyboardShortcuts.bind(bindValue)()['Mod-Shift-p']();
      expect(toggleSpoiler).toHaveBeenCalled();
    });
  });
});
