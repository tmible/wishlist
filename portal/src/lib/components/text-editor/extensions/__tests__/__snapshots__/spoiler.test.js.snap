// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text editor / extensions / spoiler > add keyboard shortcuts > should add keyboard shortcuts 1`] = `
{
  "Mod-Shift-p": [Function],
}
`;

exports[`text editor / extensions / spoiler > should be defined 1`] = `
Mark {
  "child": null,
  "config": {
    "addCommands": [Function],
    "addKeyboardShortcuts": [Function],
    "defaultOptions": {},
    "name": "spoiler",
    "parseHTML": [Function],
    "renderHTML": [Function],
  },
  "name": "spoiler",
  "options": {},
  "parent": null,
  "storage": {},
  "type": "mark",
}
`;

exports[`text editor / extensions / spoiler > should parse HTML 1`] = `
[
  {
    "tag": "span[data-entity="spoiler"]",
  },
]
`;

exports[`text editor / extensions / spoiler > should render HTML 1`] = `
[
  "span",
  {
    "class": "spoiler text-transparent hover:text-base-content",
  },
  0,
]
`;
